numba/typed_passes.py

```diff
@@ -473,10 +479,44 @@ def run_pass(self, state):
 
         return True
 
-    def _do_work(self, state, work_list, block, i, expr):
-        from numba.inline_closurecall import (inline_closure_call,
-                                              callee_ir_validator)
+    def _do_work_getattr(self, state, work_list, block, i, expr):
+        recv_type = state.type_annotation.typemap[expr.value.name]
+        recv_type = types.unliteral(recv_type)
+        matched = state.typingctx.find_matching_getattr_template(
+            recv_type, expr.attr,
+        )
+        if not matched:
+            return False
+        template = matched['template']
+        if getattr(template, 'is_method', False):
+            # The attribute template is representing a method.
+            # Don't inline the getattr.
+            return False
+
+        inline_type = getattr(template, '_inline', None)
+        if inline_type is None:
+            # inline not defined
+            return False
+        sig = typing.signature(matched['return_type'], recv_type)
+        arg_typs = sig.args
+
+        if not inline_type.is_never_inline:
+            try:
+                impl = template._overload_func(recv_type)
+                if impl is None:
```

Comment:
This construct strikes me as somewhat odd, am I correct in guessing that `template._overload_func` could also raise and that is why this is constructed this way?

Reply:
The `_overload_func` corresponds to the decorated function by a `@overload*`. It can return None to indicate it does match or raise an exception to indicate error; i.e. either typing or implementation error. In any case, the logic here say it cannot cause the inlining to error out.

Reply:
:+1: